src/main/scala/com/gu/subscriptions/cas/service/SubscriptionService.scala

```diff
@@ -0,0 +1,114 @@
+package com.gu.subscriptions.cas.service
+
+import com.amazonaws.regions.{Regions, Region}
+import com.gu.membership.zuora.ZuoraApiConfig
+import com.gu.membership.zuora.soap.Zuora._
+import com.gu.membership.zuora.soap._
+import com.gu.monitoring.{CloudWatch, ZuoraMetrics}
+import com.gu.subscriptions.cas.config.Configuration
+import com.gu.subscriptions.cas.model.SubscriptionExpiration
+import com.gu.subscriptions.cas.service.utils.ScheduledTask
+import com.typesafe.scalalogging.LazyLogging
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.concurrent.duration._
+
+class SubscriptionService(zuoraClient: ZuoraClient,
+                          knownProducts: List[String],
+                          cloudWatch: CloudWatch) extends LazyLogging {
+  import spray.json._
+  import DefaultJsonProtocol._
+
+  case class SubsRequest(subscriberId: Option[String], password:String)
+  implicit val subsRequestFormat = jsonFormat2(SubsRequest)
+
+  def extractZuoraSubscriptionId(subsRequest: SubsRequest): Option[String] =
+    subsRequest.subscriberId.filter(_.startsWith("A-S"))
+
+  def extractSubsRequest(requestJson: String): SubsRequest = {
```

Comment:
We're going to respond with an error message in JSON format. See https://github.com/guardian/content-authorisation-proxy/commit/8ee19beb173071e0626a2bb24cf504ceda0b2156